Cargo.toml

```diff
@@ -35,35 +35,35 @@ default = [
     "websocket",
     "yamux",
 ]
-autonat = ["libp2p-autonat"]
-dcutr = ["libp2p-dcutr", "libp2p-metrics/dcutr"]
-deflate = ["libp2p-deflate"]
-dns-async-std = ["libp2p-dns", "libp2p-dns/async-std"]
-dns-tokio = ["libp2p-dns", "libp2p-dns/tokio"]
-floodsub = ["libp2p-floodsub"]
-identify = ["libp2p-identify", "libp2p-metrics/identify"]
-kad = ["libp2p-kad", "libp2p-metrics/kad"]
-gossipsub = ["libp2p-gossipsub", "libp2p-metrics/gossipsub"]
-metrics = ["libp2p-metrics"]
-mdns = ["libp2p-mdns"]
-mplex = ["libp2p-mplex"]
-noise = ["libp2p-noise"]
-ping = ["libp2p-ping", "libp2p-metrics/ping"]
-plaintext = ["libp2p-plaintext"]
-pnet = ["libp2p-pnet"]
-relay = ["libp2p-relay", "libp2p-metrics/relay"]
-request-response = ["libp2p-request-response"]
-rendezvous = ["libp2p-rendezvous"]
-tcp-async-io = ["libp2p-tcp", "libp2p-tcp/async-io"]
-tcp-tokio = ["libp2p-tcp", "libp2p-tcp/tokio"]
-uds = ["libp2p-uds"]
+autonat = ["dep:libp2p-autonat"]
+dcutr = ["dep:libp2p-dcutr", "libp2p-metrics?/dcutr"]
+deflate = ["dep:libp2p-deflate"]
+dns-async-std = ["dep:libp2p-dns", "libp2p-dns?/async-std"]
```

Comment:
Technically, setting `async-std` as a weak-dependency feature would not be necessary because we activate `libp2p-dns` anyway. But I also don't think it is harmful :)